apps/src/templates/progress/progressHelpers.js

```diff
@@ -42,7 +37,33 @@ export function lessonIsVisible(lesson, state, viewAs) {
 }
 
 /**
- * Check to see if a stage/lesson is locked for all stages in the current section
+ * Treat the lesson as locked if either
+ * (a) it is locked for this user (in the case of a student)
+ * (b) non-verified teacher
+ * (c) signed out user
+ * @param {number} lesson - the lesson we're querying
+ * @param {object} state - State of our entire redux store
+ * @param {ViewType} viewAs - Are we interested in whether the lesson is viewable
+ *   for students or teachers
+ * @returns {boolean} True if the provided lesson is visible
+ */
+export function lessonIsLockedForUser(lesson, levels, state, viewAs) {
+  if (!lesson.lockable) {
+    return false;
+  }
+  // Signed out user
+  if (!state.currentUser.userId) {
+    return true;
+  }
+  // non-verified teacher
+  if (viewAs === ViewType.Teacher && !state.stageLock.lockableAuthorized) {
```

Comment:
I think if the user is a teacher and the lockable is authorized, we always want to return `false` right?  Put another way:
- For a teacher, whether or not the lessonIsLockedForUser depends on lockableAuthorized
- For a student, whether or not the lessonIsLockedForUser depends on stageLocked(levels)

Reply:
Ah I see what you are getting it. I can update the method to make that more clear